server/app/modules/SecurityModule.java

```diff
@@ -120,20 +120,39 @@ protected void configure() {
         new PlayCookieSessionStore(new ShiroAesDataEncrypter(aesKey));
     bind(SessionStore.class).toInstance(sessionStore);
 
-    bindAdminIdpProvider();
+    bindAdminIdpProvider(configuration);
     bindApplicantIdpProvider(configuration);
   }
 
-  private void bindAdminIdpProvider() {
-    // Currently the only supported admin auth provider. As we add other admin auth providers,
-    // this can be converted into a switch statement.
-    bind(IndirectClient.class)
-        .annotatedWith(AdminAuthClient.class)
-        .toProvider(AdfsClientProvider.class);
+  private void bindAdminIdpProvider(com.typesafe.config.Config config) {
+    AuthIdentityProviderName idpName =
+        AuthIdentityProviderName.adminIdentityProviderfromConfig(config);
+    try {
+      switch (idpName) {
+        case ADFS_ADMIN:
+          bind(IndirectClient.class)
+              .annotatedWith(AdminAuthClient.class)
+              .toProvider(AdfsClientProvider.class);
+          logger.info("Using ADFS for admin auth provider");
+          break;
+        case GENERIC_OIDC_ADMIN:
+          bind(IndirectClient.class)
+              .annotatedWith(AdminAuthClient.class)
+              .toProvider(auth.oidc.admin.GenericOidcClientProvider.class);
+          logger.info("Using Generic OIDC for admin auth provider");
+          break;
+        default:
+          throw new ConfigurationException("Unable to create admin identity provider: " + idpName);
+      }
+    } catch (RuntimeException e) {
+      logger.error("Error getting admin auth provider");
+      throw e;
```

Comment:
**question (blocking):** when do we expect a RuntimeException with this code? Is the `try/catch` necessary here?